numpy/typing/__init__.py

```diff
@@ -0,0 +1,3 @@
+from ._array_like import _SupportsArray, ArrayLike
```

Comment:
There isn't much code in this package, but since typing is so verbose it would be a little painful to keep `Any`, `overload`, ... out of the public namespace if we crammed everything in a `typing.py` file. So instead make a package and use this init to make sure we're only exporting exactly what we mean to.